lib/gollum-lib/filter/emoji.rb

```diff
@@ -0,0 +1,23 @@
+# ~*~ encoding: utf-8 ~*~
+
+# Emoji
+#
+# Render emoji such as :smile:
+class Gollum::Filter::Emoji < Gollum::Filter
+  EMOJI_TAG_PATTERN = %r{
+    (?<!\[{2})
+    :(?:([\w-]|</?em>)+):
+    (?!\]{^2})
+  }ix
+
+  def extract(data)
+    data
+  end
+
+  def process(data)
+    data.gsub EMOJI_TAG_PATTERN do |tag|
+      name = tag[1..-2].gsub(%r{</?em>}, '_')   # undo emphasis
```

Comment:
> On 23 mei 2016, at 15:11, Sven Schwyn <notifications@github.com> wrote:
> 
> Of course, this collision could be solved properly elsewhere, but since emoji are an opt-in feature, I figured this should be taken care of in the context of the emoji even if it appears a little hackish.

I agree with that approach. Just mind that, since the filter chain is customizable, we cant assume that the markdown parser has already kicked in. No problem for this code, just good to keep in mind that users *can* possibly break this functionality if they're not careful. :)